diff --git a/docs/content/manual/manual.yml b/docs/content/manual/manual.yml
index bfbde89..cfa5b8a 100644
--- a/docs/content/manual/manual.yml
+++ b/docs/content/manual/manual.yml
@@ -1922,7 +1922,9 @@ sections:
           * `@uri`:
 
             Applies percent-encoding, by mapping all reserved URI
-            characters to a `%XX` sequence.
+            characters to a `%XX` sequence, as specified by RFC 3986.
+            The unreserved characters `A-Z`, `a-z`, `0-9`, `-`, `.`,
+            `_` and `~` are left as is.
 
           * `@csv`:
 
diff --git a/jq.1.prebuilt b/jq.1.prebuilt
index 82623e1..c5435f7 100644
--- a/jq.1.prebuilt
+++ b/jq.1.prebuilt
@@ -2076,7 +2076,7 @@ Applies HTML/XML escaping, by mapping the characters \fB<>&\'"\fR to their entit
 \fB@uri\fR:
 .
 .IP
-Applies percent\-encoding, by mapping all reserved URI characters to a \fB%XX\fR sequence\.
+Applies percent\-encoding, by mapping all reserved URI characters to a \fB%XX\fR sequence, as specified by RFC 3986\. The unreserved characters \fBA\-Z\fR, \fBa\-z\fR, \fB0\-9\fR, \fB\-\fR, \fB\.\fR, \fB_\fR and \fB~\fR are left as is\.
 .
 .TP
 \fB@csv\fR:
diff --git a/src/builtin.c b/src/builtin.c
index 9b2d9a2..79e9636 100644
--- a/src/builtin.c
+++ b/src/builtin.c
@@ -642,7 +642,7 @@ static jv f_format(jq_state *jq, jv input, jv fmt) {
     input = f_tostring(jq, input);
 
     int unreserved[128] = {0};
-    const char* p = CHARS_ALPHANUM "-_.!~*'()";
+    const char* p = CHARS_ALPHANUM "-_.~";
     while (*p) unreserved[(int)*p++] = 1;
 
     jv line = jv_string("");
diff --git a/tests/jq.test b/tests/jq.test
index ca8e270..5597d63 100644
--- a/tests/jq.test
+++ b/tests/jq.test
@@ -68,7 +68,7 @@ null
 "1,\"<>&'\"\"\t\""
 "1\t<>&'\"\\t"
 "&lt;&gt;&amp;&apos;&quot;\t"
-"%3C%3E%26'%22%09"
+"%3C%3E%26%27%22%09"
 "'<>&'\\''\"\t'"
 "PD4mJyIJ"
 "<>&'\"\t"
@@ -86,6 +86,14 @@ null
 "\u03bc"
 "%CE%BC"
 
+@uri
+"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_.~"
+"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_.~"
+
+@uri
+" !\"#$%&'()*+,/:;<=>?@[\\]^`{|}"
+"%20%21%22%23%24%25%26%27%28%29%2A%2B%2C%2F%3A%3B%3C%3D%3E%3F%40%5B%5C%5D%5E%60%7B%7C%7D"
+
 @html "<b>\(.)</b>"
 "<script>hax</script>"
 "<b>&lt;script&gt;hax&lt;/script&gt;</b>"
